src/cmd/flash.js

```diff
@@ -301,4 +311,43 @@ module.exports = class FlashCommand extends CLICommandBase {
 		}
 		return Array.from(binaries);
 	}
+
+	async _getDeviceOsBinaries({ skipDeviceOSFlash, target, files, firmwareVersion, platformId, applicationOnly }) {
+		const { particleApi } = this._particleApi();
+		const [binary] = files;
```

Comment:
You need to find the application module, not just pick the first file.

Reply:
Did it here! https://github.com/particle-iot/particle-cli/pull/657/commits/45be6c14118f41cf918575c8b02679d83783bf2d.